tests/xharness/Xharness.Tests/Logging/Tests/CallbackLogTest.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using NUnit.Framework;
+using Xharness.Logging;
+
+namespace Xharness.Tests.Logging.Tests {
+
+	[TestFixture]
+	public class CallbackLogTest {
+
+		[Test]
+		public void OnWriteTest ()
+		{
+			var message = "This is a log message";
+			bool called = false;
+			string data = null;
+
+			Action<string> cb = (d) => {
+				called = true;
+				data = d;
+			};
+
+			var log = new CallbackLog (cb);
+			log.Write (message);
+			Assert.IsTrue (called, "called");
```

Comment:
FYI: The assert message is in case of a failed assert, so here it would be `"Callback was not called"`

Reply:
We use them to identify what assert failed, but is not bad to add a more decent message.